### ISSUE TITLE
Single Stat Graph JavaScript Error in Safari

### ISSUE THREAD

**Issue by gotjosh** [grafana/grafana#18237]
**What happened**:

There is a JavaScript error that appears on a Single Stat panel in Safari, but not in Chrome. It reads: `toFixed() argument must be between 0 and 20`.

Range to text mapping specified alongside a decimal value on a single stat graph. 

**What you expected to happen**:

No JavaScript errors on any browser.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a single stat panel
2. Add a range to text mapping (e.g. from 5 to 500000 with the text OK)
3. Add a decimal value (e.g. 40)

No error should be displayed on chrome but it does show "no value" on Safari.

**Environment**:

TBD

- Grafana version: Latest
- Data source type & version: Tested with Splunk but I suspect this will affect any data source
- Tested on Chrome Version 75.0.3770.142 (Official Build) (64-bit) and Safari Version 12.1.1 (14607.2.6.1.1)

**Comment by ryantxu** [comment]
@gotjosh -- ask for their panel config JSON.  The `toFixed` is most likely called in the unit conversion/formatting section

**Comment by torkelo** [comment]
if we had their unit & the query response we could probably replicate it.

**Comment by gotjosh** [comment]
Thanks for the tip @torkelo and @ryantxu. I got a hold of the panel config and noticed the problem. I have updated the issue with the details on how to reproduce.

I'm not extensible familiar with these type of issues on the front-end and a couple of questions pop to my head:

- Arguably, we could classify this as an input error. Removing the decimal value does seem to "fix" the issue, but I'm thinking we could do more in terms of validation and feedback for the user. How do we reach a consensus on how to best move forward? 

- How are these issues typically handled? Do we tell the customer to do X instead of Y (in this case remove the decimal value) or we add the fixes/validations in place and tell them to upgrade or maybe even both.

- Is there a world where don't do anything and classify this as a "misuse"?
